### DISCUSSION TITLE
APISIX gateway GRPC transcode returning static token

### DISCUSSION THREAD

**Discussion by ochuko3d** [apache/apisix#7616] [Q&A]
Hi guys,

I would really love help resolving an issue around  grpc transcode returning static tokens. My proto file has a an import embedded. import proto and was able to upload it to apisix admin,  and doing a trial run i keep getting an empty static token. 

APISIX log report 200 to the server but its not responding properly

**Comment by tzssangglass** [comment]
Can you provide the reproduction steps?

**Comment by tokers** [comment]
@ochuko3d Could you describe your issue elaborately? I cannot capture the meaning.

**Comment by ochuko3d** [comment]
@tokers @tzssangglass 

**Reproduction steps**
I have tried to connect to our internal pod running a Grpc application on kubernetes using the APISIX ingress controller + gateways. I have the dashboard to ensure i see the configuration done.

I followed the steps giving in the following API6 doc: https://apisix.apache.org/docs/apisix/plugins/grpc-transcode/
My proto file has the _import function_ which import an external proto file so i had to create a **.pb** file. i passed the file true base64 **cat proto.pb | base64**, then pushed it to the admin by using the **curl** function.

I mapped the transcode on the route as below

"plugins": {
        "grpc-transcode": {
            "proto_id": "1",
            "service": "login.Account.Service",
            "method": "login"
        }

After doing that the route is create but when we carry out a test using postman, nothing is returned

**We are expecting to receive the below response**,
{
	"token": "74r347yr834y734...",
    "user": {
      "firName": "name",
      "lastName": "vlid"
      ...
    }
}
But we are receiving this,
{
	"token": ""
}.

Upon checking the api6 pod i can see status code 200, but when I check the application on the pod, it seems apisix doesn't talk with the pod as no record is seen on the pod.

**observation**
1. If we switch to a NodePort, and permit the port on AWS security group, we get records on the application pod and the test works appropriately .
2. I noticed that kong api-gateway has an annotation to include in the service yaml for Grpc plugin on the cluster, is there a similar annotation or requirement for Kubernetes cluster.
3. We can reach other HTTP protocols like argocd on 8080, running in the cluster but not the Grpc application

**Reply by tokers** [reply]
@ochuko3d Try to see https://apisix.apache.org/docs/apisix/plugins/grpc-transcode/#enabling-the-plugin, it may help you.

**Comment by ochuko3d** [comment]
Thanks for the prompt response @tokers but o have tried that, the only difference would be adding the Pb_option. Do i need to modify the kubernetes service object

**Reply by tokers** [reply]
So you already use the python script to generate the pb file?

**Reply by tokers** [reply]
What's the upstream configuration?
